torch/csrc/autograd/profiler.cpp

```diff
@@ -274,6 +288,66 @@ struct ProfilerThreadLocalState
   }
 
  private:
+  void setSourceLocation(Event& evt, const at::RecordFunction& fn) {
+    std::lock_guard<std::mutex> guard(state_mutex_);
+    if (fn.scope() == at::RecordScope::BACKWARD_FUNCTION) {
+      // if that's a backward function then take the source location
+      // from the corresponding function
+      auto fwd_pair = std::make_pair(fn.seqNr(), fn.forwardThreadId());
+      auto it = fwd_source_locs_.find(fwd_pair);
+      if (it != fwd_source_locs_.end()) {
+        evt.setSourceLocation(it->second);
+      }
+      // in some cases we won't find the corr. forward node (e.g. GraphRoot)
+    } else {
+      auto ts_loc = torchScriptCallSite();
```

Comment:
please leave a comment here regarding under which conditions ts_loc and py_stack are necessary (ts_loc for scripted, py_stack for eager, right?), and try not to call py_stack in scripted mode